API fix params validation in SGD inherited models (#20683)
scikit-learn/scikit-learn: doc/whats_new/v1.0.rst

  coordinate descent solver. Otherwise, an error will be raised.
  :pr:`19391` by :user:`Shao Yang Hong <hongshaoyang>`.

> - |API| Force the validation to be made at `fit` for the following estimators
>   to follow our scikit-learn convention:

Reviewer: To specify that validation is not done in `__init__` anymore.

```suggestion
- |API| Keyword validation has moved from `__init__` and `set_params` to `fit`
   for the following estimators conforming to scikit-learn's conventions:
```

  :class:`linear_model.SGDClassifier`,
  :class:`linear_model.SparseSGDClassifier`,
  :class:`linear_model.SGDRegressor`,
  :class:`linear_model.SparseSGDRegressor`,
  :class:`linear_model.SGDOneClassSVM`,
  :class:`linear_model.SparseSGDOneClassSVM`,
  :class:`linear_model.PassiveAggressiveClassifier`,
  :class:`linear_model.PassiveAggressiveRegressor`.

>   :pr:`20683` by :user:`Guillaume Lemaitre <glemaitre>`.

Reviewer 2: you could also do 

```suggestion
  :pr:`20683` by `Guillaume Lemaitre`_.
```
